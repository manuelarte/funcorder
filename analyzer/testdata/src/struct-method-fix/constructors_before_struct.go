package struct_method_fix

func NewOtherMyStruct() (m *MyStruct) {
	m = &MyStruct{Name: "John"}
	return
}

func NewMyStruct() *MyStruct {
	return &MyStruct{Name: "John"}
}

func MustMyStruct() *MyStruct {
	return NewMyStruct()
}

type MyStruct struct {
	Name string
}

func (m MyStruct) lenName() int { // want `unexported method \"lenName\" for struct \"MyStruct\" should be placed after the exported method \"SetName\"`
	return len(m.Name)
}

func (m MyStruct) GetName() string {
	return m.Name
}

func (m *MyStruct) SetName(name string) {
	m.Name = name
}
