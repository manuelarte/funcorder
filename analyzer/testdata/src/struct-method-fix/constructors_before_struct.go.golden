package struct_method_fix

func NewOtherMyStruct() (m *MyStruct) {
	m = &MyStruct{Name: "John"}
	return
}

func NewMyStruct() *MyStruct {
	return &MyStruct{Name: "John"}
}

func MustMyStruct() *MyStruct {
	return NewMyStruct()
}

type MyStruct struct {
	Name string
}

func (m MyStruct) GetName() string {
	return m.Name
}

func (m *MyStruct) SetName(name string) {
	m.Name = name
}

func (m MyStruct) lenName() int {
	return len(m.Name)
}
